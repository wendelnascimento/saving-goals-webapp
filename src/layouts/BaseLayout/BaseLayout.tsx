import React from 'react';

import Header from '../../components/Header';

import { Wrapper } from './styles';

const BaseLayout: React.FC = ({ children }) => (
  <Wrapper>
    <Header />
    {children}
  </Wrapper>
);

export default BaseLayout;
