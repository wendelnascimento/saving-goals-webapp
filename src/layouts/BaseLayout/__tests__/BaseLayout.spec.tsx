import React from 'react';
import { shallow, mount } from 'enzyme';

import BaseLayout from '../BaseLayout';

describe('<BaseLayout />', () => {
  it('Should render', () => {
    shallow(<BaseLayout />);
  });

  it('Should render with children', () => {
    shallow(
      <BaseLayout>
        <h1>Test</h1>
      </BaseLayout>,
    );
  });

  it('Should math stored snapshot', () => {
    const wrapperWithChildren = mount(
      <BaseLayout>
        <h1>Test</h1>
      </BaseLayout>,
    );
    const emptyWrapper = mount(<BaseLayout />);

    expect(wrapperWithChildren).toMatchSnapshot();
    expect(emptyWrapper).toMatchSnapshot();
  });
});
