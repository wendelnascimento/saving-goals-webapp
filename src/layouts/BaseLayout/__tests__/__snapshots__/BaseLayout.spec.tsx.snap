// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BaseLayout /> Should math stored snapshot 1`] = `
.c2 {
  width: 100px;
}

.c1 {
  max-width: 100vw;
  width: 100%;
  padding: 16px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  border-bottom: 1px solid;
}

.c0 {
  max-width: 100vw;
  min-height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<BaseLayout>
  <styled.div>
    <div
      className="c0"
    >
      <Header>
        <styled.header>
          <header
            className="c1"
          >
            <Logo>
              <styled.img
                src="test-file-stub"
              >
                <img
                  className="c2"
                  src="test-file-stub"
                />
              </styled.img>
            </Logo>
          </header>
        </styled.header>
      </Header>
      <h1>
        Test
      </h1>
    </div>
  </styled.div>
</BaseLayout>
`;

exports[`<BaseLayout /> Should math stored snapshot 2`] = `
.c2 {
  width: 100px;
}

.c1 {
  max-width: 100vw;
  width: 100%;
  padding: 16px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  border-bottom: 1px solid;
}

.c0 {
  max-width: 100vw;
  min-height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<BaseLayout>
  <styled.div>
    <div
      className="c0"
    >
      <Header>
        <styled.header>
          <header
            className="c1"
          >
            <Logo>
              <styled.img
                src="test-file-stub"
              >
                <img
                  className="c2"
                  src="test-file-stub"
                />
              </styled.img>
            </Logo>
          </header>
        </styled.header>
      </Header>
    </div>
  </styled.div>
</BaseLayout>
`;
